packages/js/data/src/countries/resolvers.ts

```diff
@@ -58,6 +60,49 @@ export function* getCountries() {
 	}
 }
 
+export function* getCurrencies() {
+	const getPriceFormat = ( symbolPosition: string ) => {
+		switch ( symbolPosition ) {
+			case 'left':
+				return '%1$s%2$s';
+			case 'right':
+				return '%2$s%1$s';
+			case 'left_space':
+				return '%1$s %2$s';
+			case 'right_space':
+				return '%2$s %1$s';
+		}
+
+		return '%1$s%2$s';
+	};
+
+	try {
+		const url = 'wc/v3/data/currencies?include_extra=true';
+		const results = yield apiFetch( {
+			path: url,
+			method: 'GET',
+		} );
+
+		return getCurrenciesSuccess(
+			results.reduce( ( acc, currency ) => {
+				acc[ currency.code ] = {
+					code: currency.code,
+					symbol: currency.symbol,
+					symbolPosition: currency.currency_position,
+					decimalSeparator: currency.number_of_decimals,
```

Comment:
```suggestion
					decimalSeparator: currency.decimal_separator,
```

<img width="1701" alt="image" src="https://github.com/user-attachments/assets/4d449301-e1b7-4a20-be92-a11b4c9df56d" />